lib/chef/application.rb

```diff
@@ -198,16 +204,19 @@ def run_application
 
     # Initializes Chef::Client instance and runs it
     def run_chef_client(specific_recipes = [])
+      unless specific_recipes.respond_to?(:size)
```

Comment:
I would use `specific_recipes.kind_of?(Array)` here and raise and raise an error if something other than an array is provided.  Otherwise we could be hiding an error from callers of this.